src/python/pants/core/goals/update_build_files_test.py

```diff
@@ -184,6 +187,58 @@ def test_black_config(black_rule_runner: RuleRunner) -> None:
     assert Path(black_rule_runner.build_root, "BUILD").read_text() == "tgt(name='t')\n"
 
 
+# ------------------------------------------------------------------------------------------
+# Yapf formatter fixer
+# ------------------------------------------------------------------------------------------
+
+
+@pytest.fixture
+def yapf_rule_runner() -> RuleRunner:
+    return RuleRunner(
+        rules=(
+            format_build_file_with_yapf,
+            update_build_files,
+            *pants_bin.rules(),
+            *config_files.rules(),
+            *pex.rules(),
+            SubsystemRule(Yapf),
+            SubsystemRule(UpdateBuildFilesSubsystem),
+            UnionRule(RewrittenBuildFileRequest, FormatWithYapfRequest),
+        )
+    )
+
+
+def test_yapf_fixer_fixes(yapf_rule_runner: RuleRunner) -> None:
+    yapf_rule_runner.write_files({"BUILD": "tgt( name =  't' )"})
+    result = yapf_rule_runner.run_goal_rule(UpdateBuildFilesGoal, env_inherit=BLACK_ENV_INHERIT)
+    assert result.exit_code == 0
+    assert result.stdout == dedent(
+        """\
+        Updated BUILD:
+          - Format with Yapf
+        """
+    )
+    assert Path(yapf_rule_runner.build_root, "BUILD").read_text() == 'tgt(name="t")\n'
```

Comment:
I don't understand how the MacOS tests pass when I see  ` - Format with Black` in the output and not ``- Format with Yapf :(

Reply:
Hm. Does it work outside of tests?

Reply:
Do you mean if I run the test alone? No, I am still getting an empty string as the `result.stdout` when running locally.

Reply:
I mean if you run `./pants update-build-files --formatter=yapf` in the real world, not in a test

Reply:
Oh, I've posted this in the body description, sorry if that wasn't too easy to notice!

Running on files that would be formatted:

```
$ ./pants update-build-files --fmt --formatter=yapf
Updated src/python/pants/backend/python/lint/yapf/BUILD:
  - Format with Black
```

The formatting does happen, and yapf.pex was indeed built (according to the messages), but unfortunately, the message still shows `- Format with Black` for some reason (so it could be formatted by Black as well).

Reply:
Ah, no I just didn't read with enough attention to detail. The other thread in this review should fix that.

Reply:
It works now with Yapf! :) 

```
./pants update-build-files --fmt --formatter=yapf
Updated src/python/pants/core/BUILD:
  - Format with Yapf
```

I had to explicitly add arguments "--in-place" for it to modify BUILD files in-place; it has previously only put the formatted file to the stdout :D